docs/design/sql/05-ddl.md

```diff
@@ -0,0 +1,264 @@
+```
+
+### 1.2 Description
+
+`CREATE EXTERNAL TABLE` registers a virtual table that references an external storage system. For some storage systems
+it does not create a physical entity until a write occurs. Column list is optional; if not specified, we will attempt
+to infer it from sample data or from the options.
+
+The table is created in the `public` schema. The name of the table must be distinct from the name of any other table in
+the schema.
+
+### 1.3 Parameters
+
+**OR REPLACE**
+
+Replace table definition if one with the same name already exists. Atomically, drops and creates a reference to the
+external storage system, as a result no physical entity or data is being deleted though.
+
+**IF NOT EXISTS**
+
+Take no action if a table with the same name already exists.
+
+**table_name**
+
+The name of the table to be created.
+
+**column_name**
+
+The name of the column in the table.
+
+**column_type**
+
+The data type of the column. For more information on the supported data types, refer to [type system document](01-type-system.md).
+
+**column_external_name**
+
+The path to the field in the external storage system. Used to link the column to a physical field with a different name.
+It's optional, the default value is specific for each table type. For example for `IMap` it defaults to `this.<columnName>`
+where `this` refers to the entry value.
+
+**table_type**
+
+The external storage system identifier. Supported table types are listed in Table 1. Not all storage systems implement
+full set of operations, so for instance you might be able to `SELECT` from a given table but not `INSERT` into it.
+
+*Table 1: CREATE EXTERNAL TABLE supported table types*
+
+| Table type | Description |
+|---|---|
+| IMap | Table backed by Hazelcast `IMap` |
+| ReplicatedMap | Table backed by Hazelcast `ReplicatedMap` |
+| File | Table backed by files in a directory |
+
+**table_parameter**
+
+Parameter specific to the external storage system - connection string, serialization format etc.
+This is an identifier, not a string. Therefore it must not be quoted as a string. It can be quoted in double
+quotes if it contains special characters (such as space, period...).
+
+**table_parameter_value**
+
+Parameter value as a string literal.
+
+### 1.4 Examples
+
+- Create an external table referencing an `IMap` with inferred schema.
+
+  ```sql
+  CREATE EXTERNAL TABLE persons
+  TYPE "com.hazelcast.IMap"
+  OPTIONS (
+    "serialization.key.format" 'java',
+    "serialization.key.java.class" 'java.lang.Integer',
+    "serialization.value.format" 'java',
+    "serialization.value.java.class" 'java.lang.String'
+  )
+  ```
+
+  Column names and types are inferred as:
+
+  ```sql
+  (
+    __key INT,
+    this VARCHAR
+  )
+  ```
+
+- Create an external table referencing an `IMap` with explicit schema.
+
+  ```sql
+  CREATE EXTERNAL TABLE persons (
+    id INT EXTERNAL NAME __key,
+    name VARCHAR
+  )
+  TYPE "com.hazelcast.IMap"
```

Comment:
Have we agreed to use short names for connectors? If yes, then this part of the document should be adjusted (probably some other places as well)